app/feed/page.tsx

```diff
@@ -1,21 +1,41 @@
 import LoadingText from "@/components/LoadingText";
 import { IGitHubEvent } from "@/lib/gh_events";
-import GitHubEvent from "@/components/gh_events/GitHubEvent";
 import { env } from "@/env.mjs";
 import octokit from "@/lib/octokit";
+import {
+  // fetchAllBranchName,
+  fetchAllReposName,
+} from "../api/leaderboard/functions";
+import GithubFeedFilter from "../../components/gh_events/GithubFeedFilter";
+import GitHubEvent from "@/components/gh_events/GitHubEvent";
+import { FilterOption } from "@/lib/types";
 
 const GITHUB_ORG: string = env.NEXT_PUBLIC_GITHUB_ORG;
 
 export const revalidate = 600;
 
-type Props = {
-  searchParams: {
-    page?: string;
-  };
-};
-
-export default async function FeedPage({ searchParams }: Props) {
-  const events = await octokit.paginate(
+export default async function FeedPage({ searchParams }: any) {
+  const { Repository, Events } = searchParams;
+  const repositories = await fetchAllReposName();
+  const filterEvents: FilterOption[] = [
+    { title: "Repository", options: ["All", ...repositories] },
```

Comment:
use `null`/`undefined` to cover the case for "All" instead of adding it as a valid value.

```suggestion
    { title: "Repository", options: [repositories] },
```